### PR TITLE
Handle case where lastLogcatTimestamp is null

### PR DESCRIPTION
**Pull request by blasten**
## Description

`flutter run` fails if `lastLogcatTimestamp` is `null`. This happens when the device logs are empty.

## Tests

I added the following tests: Unit test

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require Flutter developers to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (Please read [Handling breaking changes]). *Replace this with a link to the e-mail where you asked for input on this proposed change.*
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter_tools/lib/src/android/android_device.dart [hunk 1]
```diff
@@ -1009,7 +1009,7 @@ class _AdbLogReader extends DeviceLogReader {
   void _start() {
     final String lastTimestamp = device.lastLogcatTimestamp;
     // Start the adb logcat process and filter the most recent logs since `lastTimestamp`.
-    final List<String> args = <String>['logcat', '-v', 'time', '-T', lastTimestamp];
+    final List<String> args = <String>['logcat', '-v', 'time', '-T', lastTimestamp ?? ''];
```
Comment (zanderso) [inline packages/flutter_tools/lib/src/android/android_device.dart]: How about:
```dart
final List<String> args = <String>[
  'logcat',
  '-v',
  'time',
  if (lastTimestamp != null) ...<String>[
    '-T',
    lastTimestamp,
  ],
];
```

**Review by zanderso — APPROVED** [review]
lgtm w/ comment discussed offline.

File: packages/flutter_tools/lib/src/android/android_device.dart (hunk 1 above)
Comment (blasten) [reply packages/flutter_tools/lib/src/android/android_device.dart]: I will update this as I find out more.

**Comment by blasten** [comment]
Merging on red of fix some of the devicelab tests.

**Pull request merged**